observing_examples.ipynb

```diff
@@ -0,0 +1,997 @@
+    "bandpasses = {}\n",
+    "for band in sdss:\n",
+    "    local_path = download_file(svo_sdss_link + band + '_filter')\n",
+    "    bp = SpectralElement.from_file(local_path)\n",
+    "    bandpasses[band] = bp\n",
+    "    \n",
+    "    plt.plot(bp.waveset, bp(bp.waveset), label=band)\n",
+    "plt.legend(loc='lower right')\n",
+    "plt.xlim(3000, 14000)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=\"atmosphere\"></a>\n",
+    "## 5. Model the attenuation by the atmosphere"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Model the sky by querying results from the <a href=\"http://etimecalret-001.eso.org/observing/etc/bin/gen/form?INS.MODE=swspectr+INS.NAME=SKYCALC\">SkyCalc Sky Model Calculator</a>:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "83\n",
+      "73\n",
+      "77\n",
+      "80\n",
+      "76\n"
+     ]
+    }
+   ],
+   "source": [
+    "import json\n",
+    "import requests\n",
+    "\n",
+    "# SkyCalc recognizes Cerro Paranal by its id number 2460, which we \n",
+    "# got from the skycalc_cli source code\n",
+    "paranal_id = '2640'\n",
+    "\n",
+    "# Set up parameters for the observing night in the format that skycalc_cli would use\n",
+    "params_path = download_file('https://www.eso.org/observing/etc/' + \n",
+    "                            'doc/skycalc/skycalc_defaults.txt')\n",
+    "params = {}\n",
+    "with open(params_path) as o:\n",
+    "    for row in o.readlines():\n",
+    "        columns = row.split(':')\n",
+    "        columns = [column.strip() for column in columns]\n",
+    "        key, value = columns\n",
+    "        \n",
+    "        # convert numbered values to floats\n",
+    "        try:\n",
+    "            value = int(value)\n",
+    "        except:\n",
+    "            try:\n",
+    "                value = float(value)\n",
+    "            except:\n",
+    "                pass\n",
+    "        \n",
+    "        params[key] = value\n",
+    "        \n",
+    "        if key == \"observatory\":\n",
+    "            params['observatory'] = paranal_id\n",
+    "        \n",
+    "# Use the bit from skycalc_cli which queries from the SkyCalc Sky Model\n",
+    "server = 'http://etimecalret-001.eso.org'\n",
+    "url = server + '/observing/etc/api/skycalc'\n",
+    "response = requests.post(url, data=json.dumps(params))\n",
+    "results = json.loads(response.text)\n",
+    "\n",
+    "status = results['status']\n",
+    "tmpdir = results['tmpdir']\n",
+    "tmpurl = server + '/observing/etc/tmp/' + \\\n",
+    "         tmpdir + '/skytable.fits'\n",
+    "\n",
+    "if status == 'success':\n",
+    "    try: \n",
+    "        response = requests.get(tmpurl, stream=True)\n",
+    "        data = response.content\n",
```

Comment:
Write the binary content to a FITS file and try to read it back with `fits.open`. To write, maybe try:

```python
fits.writeto('skytable.fits', data, overwrite=True)
```

but I am not 100% sure.

Reply:
Ah okay, so there's not a nice way to do this without writing to a file?

Reply:
If there is one, I don't remember. Do you want to ask on Astropy Slack in `#fits` channel?

Reply:
You can also try `fits.open()` directly on `data` but I am also not 100% sure that would work.

Reply:
No luck there. Maybe creating a file is okay?

Reply:
Probably okay, as long as:

1. Your code automatically overwrites without raising an error when you run it multiple times.
2. You exclude the file from version control in `.gitignore`.